docs/tutorials/stored-proc-tutorial.md

```diff
@@ -0,0 +1,89 @@
+---
+sidebar_position: 5
+title: Stored Procedures
+description: Learn how to improve database interaction performance with Stored Procedures
+---
+
+In this guide, you'll learn how to interact with your database using stored procedures.
+
+Stored procedures are similar to [transaction functions](./transaction-tutorial.md). 
+They are used to perform operations on your application database.
+However, stored procedure functions run inside the database where transaction functions run on the application server. 
+Running your business logic inside the database improves both performance and scalability. 
+Transaction functions require at least three network round trips to the database in addition to any queries executed by the transaction function itself.
+Stored procedure functions only require a single network round trip to the database, while still providing the same behavior and guarantees as transaction functions.
+
+While most database management systems provide support for stored procedures, they are often avoided because they are hard to use.
+They typically need to be written in a custom language such as [PL/pgSQL](https://www.postgresql.org/docs/current/plpgsql.html).
+Additionally, they are not usually integrated into the application development process.
+DBOS stored procedure functions, in contrast, are written in TypeScript like the rest of your DBOS application.
+Additionally, we provide the DBOS Compiler to stored procedure functions from your application and deploy them to your application database.
+
+Here's an example of a stored procedure function.
+You'll notice it is similar to the [example transaction function](./transaction-tutorial.md) from the transaction tutorial. 
```

Comment:
Let's switch this to the simpler example from https://github.com/dbos-inc/dbos-docs/pull/155